docs/CHANGES_4.0.0.md

```diff
@@ -1,39 +1,249 @@
+So it made sense for our library to directly expose the option validation from nodejs.
+If you need to test TLS connections without verifying server identity pass in `{ checkServerIdentity: () => {} }`.
+
+#### Kerberos / GSSAPI
+
+`gssapiServiceName` has been removed.
+Users should use authMechanismProperties.SERVICE_NAME like so:
+
+- In a URI query param: `?authMechanismProperties=SERVICE_NAME:alternateServiceName`
+- Or as an option: `{ authMechanismProperties: { SERVICE_NAME: 'alternateServiceName' } }`
+
+### db.collection no longer accepts a callback
+
+The only option that required the use of the callback was strict mode and that option can still be used in `db.createCollection`.
+The strict option would return an error if the collection does not exist.
+
+Another way to assert a collections existence is using this snippet below:
 
 ```javascript
-// Declare API version "1" for the client
-client = new MongoClient(uri, { serverApi: { version: '1' } });
+const collections = (await db.listCollections({}, { nameOnly: true })
+  .toArray())
+  .map(({name}) => name); // map to get string[]
+if (!collections.includes(myNewCollectionName)) {
+    await db.createCollection(myNewCollectionName)
+}
+```
+
+### BulkWriteError renamed to MongoBulkWriteError
+
+In 3.x we exported both the names above, we now only export `MongoBulkWriteError`.
+Users testing for `BulkWriteError`s should be sure to import the new class name `MongoBulkWriteError`.
+
+### Db no longer emits events
+
+The Db instance is no longer an EventEmitter, all events your application is concerned with can be listened to directly from the `MongoClient` instance.
 
-cursor = client.db('database').collection('coll').find(...);
+### Collection.group() removed
+
+The collection `group()` helper has been deprecated in MongoDB since 3.4 and is now removed from the Driver.
+The same functionality can be achieved using the aggregation pipeline's `$group` operator.
+
+### GridStore removed
+
+The deprecated GirdStore API has been removed from the driver.
+For more information on GridFS [see the mongodb manual](https://docs.mongodb.com/manual/core/gridfs/).
+
+Below are some snippets that represent equivalent operations:
+
+#### Construction
+
+```javascript
+// old way
+const gs = new GridStore(db, filename, mode[, options])
+// new way
+const bucket = new GridFSBucket(client.db('test')[, options])
 ```
 
-### Strict mode
+#### File seeking
 
-Declaring a `strict` API version will cause the MongoDB server to reject all commands that are not part of the declared API version. This includes command options and aggregation pipeline stages. For example, the following `find` call would fail because the `tailable` option is not part of version 1:
+Since GridFSBucket uses NodeJS Stream API you can replicate file seek-ing by using the start and end options creating a downloadStream from your GridFSBucket
```

Comment:
```suggestion
Since GridFSBucket uses NodeJS Stream API you can replicate file seek-ing by using the start and end options creating a download stream from your GridFSBucket
```